internal/util/http.go

```diff
@@ -35,10 +40,23 @@ type HttpClient struct {
 
 // NewHttpClient creates a new client with the given logger.
 func NewHttpClient(logger *log.Logger) HttpClient {
+	return NewHttpClientWithTimeouts(logger, nil, nil)
+}
+
+func NewHttpClientWithTimeouts(logger *log.Logger, responseHeader, total *int) HttpClient {
+	if responseHeader == nil {
+		tmp := defaultHttpResponseHeaderTimeout
+		responseHeader = &tmp
```

Comment:
I like to wrap this sort of stuff to make it obvious how crippled this language is.

```go
responseHeader = func(i int) *int { return &i }(defaultHttpResponseHeaderTimeout);
```